lang/ruby/lib/avro/schema.rb

```diff
@@ -33,7 +35,10 @@ class Schema
     LONG_MAX_VALUE = (1 << 63) - 1
 
     def self.parse(json_string)
-      real_parse(MultiJson.load(json_string), {})
+      json_obj = MultiJson.load(json_string)
+      schema = real_parse(json_obj, {})
+      schema.logical_type = LogicalTypes.parse(json_obj) if json_obj.is_a?(Hash)
```

Comment:
I think it might make more sense for this to be added to the real_parse method. The handling for when the json_obj is a different type is already built in and you wouldn't have to expose `:logical_type=`.